clang/lib/CIR/CodeGen/CIRGenFunction.cpp

```diff
@@ -149,16 +153,49 @@ mlir::LogicalResult CIRGenFunction::declare(Address addr, const Decl *var,
 
 void CIRGenFunction::startFunction(GlobalDecl gd, QualType returnType,
                                    cir::FuncOp fn, cir::FuncType funcType,
-                                   SourceLocation loc,
+                                   FunctionArgList args, SourceLocation loc,
                                    SourceLocation startLoc) {
   assert(!curFn &&
          "CIRGenFunction can only be used for one function at a time");
 
   fnRetTy = returnType;
   curFn = fn;
 
+  const auto *fd = dyn_cast_or_null<FunctionDecl>(gd.getDecl());
+
   mlir::Block *entryBB = &fn.getBlocks().front();
   builder.setInsertionPointToStart(entryBB);
+
+  // TODO(cir): this should live in `emitFunctionProlog
+  // Declare all the function arguments in the symbol table.
+  for (const auto nameValue : llvm::zip(args, entryBB->getArguments())) {
+    auto *paramVar = std::get<0>(nameValue);
```

Comment:
Change auto for real type? Same for lines below?

Reply:
Yeah, 'auto' is only allowed when the type is either: 1- so many type aliases deep that readability is helped, or 2- the type is 'named' directly/clear from what the RHS says.

Reply:
Yeah, I know this. I don't know how I missed these. I'll probably do this a few more times before it completely sinks in.